src/gui/layout/qgslayoutlegendwidget.cpp

```diff
@@ -808,8 +816,17 @@ void QgsLayoutLegendWidget::mMoveUpToolButton_clicked()
   }
   else // legend node
   {
-    _moveLegendNode( legendNode->layerNode(), _unfilteredLegendNodeIndex( legendNode ), -1 );
-    mItemTreeView->layerTreeModel()->refreshLayerLegend( legendNode->layerNode() );
+    // get the previous index, the one that will have to be below our index,
+    const QModelIndex prevIndex = index.siblingAtRow( index.row() - 1 );
+    if ( prevIndex.isValid() )
```

Comment:
Could you the same as above ?

Reply:
Oh yes, 😄 

Done, and rebased onto master.